codeexamples/SYCLBufferExampleExecutingCUDAKernels/StaticLibraryUse/src/main.cpp

```diff
@@ -0,0 +1,234 @@
+void HelloFromCUDAKernel( sycl::queue &vQ );
+void VectorInitialize( VectorInt_t &vVec, const Data_t vValue );
+void VectorAdditionUsingCUDA( sycl::queue &vQ, const VectorInt_t &vVecA, const VectorInt_t &vVecB, VectorInt_t &vVecSumParallel );
+void VectorAdditionUsingSYCL( sycl::queue &vQ, const VectorInt_t &vVecA, const VectorInt_t &vVecB, VectorInt_t &vVecSumParallel );
+
+// Asynchronous errors hander, catch faults in asynchronously executed code
+// inside a command group or a kernel. They can occur in a different stackframe, 
+// asynchronous error cannot be propagated up the stack. 
+// By default, they are considered 'lost'. The way in which we can retrieve them
+// is by providing an error handler function.
+auto exception_handler = []( sycl::exception_list vExceptions ) 
+{
+    for( std::exception_ptr const &e : vExceptions ) 
+    {
+        try 
+        {
+          std::rethrow_exception( e );
+        } 
+        catch( sycl::exception const &e ) 
+        {
+          std::cout << "Queue handler caught asynchronous SYCL exception:\n" << e.what() << std::endl;
+        }
+    }
+};
+
+// Defining a SYCL kernel as function object to carry out a vector add
+class KernelSYCLVectorAdd
+{
+  public:
+    using AccessorRead_t = sycl::accessor< Data_t, 1, sycl::access::mode::read, sycl::access::target::device >;
+    using AccessorWrite_t = sycl::accessor< Data_t, 1, sycl::access::mode::write, sycl::access::target::device >;
+        
+    KernelSYCLVectorAdd( AccessorRead_t vAccA, AccessorRead_t vAccB, AccessorWrite_t vAccSum )
+    : m_accA( vAccA )
+    , m_accB( vAccB )
+    , m_accSum ( vAccSum )
+    {}
+
+    // The parameter of the operator is the work item id.
+    void operator()( sycl::id< 1 > vI ) const 
+    { 
+        m_accSum[ vI ] = m_accA[ vI ] + m_accB[ vI ]; 
+    }
+
+  private:
+    // Create an accessor for each buffer with access permission: read, write or
+    // read/write. The accessor is a mean to access the memory in the buffer.
+    AccessorRead_t  m_accA;
+    AccessorRead_t  m_accB;
+    AccessorWrite_t m_accSum;
+};
+    
+// The program that enqueues CUDA kernels alongside SYCL kernels
+int main( void )
+{
+    std::cout << "Hello from this program." << std::endl;
+    
+    constexpr Data_t vecSize = 10000;
+    VectorInt_t vecA( vecSize ), vecB( vecSize ), vecSumParallel( vecSize );
+    VectorInitialize( vecA, 1 );
+    VectorInitialize( vecB, 1 );
+    VectorInitialize( vecSumParallel, 0 );
+    
+    try
+    {
+        const auto deviceSelector = sycl::default_selector_v;
+        sycl::queue q( deviceSelector, exception_handler );
+        
+        std::cout << "Running on device: " << q.get_device().get_info< sycl::info::device::name >() << "\n";
+
+        // Note the following kernels are not necessarily executed in the order show here.
+        // Kernels can execute out of order (asynchronous) if the call graph does not see any 
```

Comment:
```suggestion
        // Kernels can execute out of order if the call graph does not see any 
```
It's always asynchronous, but the ordering is determined by the dependencies and queue properties